core/src/test/java/org/elasticsearch/index/mapper/geo/GeoPointFieldMapperTests.java

```diff
@@ -282,6 +283,42 @@ public void testValidateLatLonValues() throws Exception {
         } catch (MapperParsingException e) {
 
         }
+
+        try {
+            defaultMapper.parse("test", "type", "1", XContentFactory.jsonBuilder()
+                    .startObject()
+                    .startObject("point").field("lat", "-").field("lon", 1.3).endObject()
+                    .endObject()
+                    .bytes());
+            fail();
+        } catch (MapperParsingException e) {
+            assertTrue(e.getRootCause() instanceof NumberFormatException);
+            assertTrue(e.getRootCause().toString().contains("java.lang.NumberFormatException: For input string: \"-\""));
```

Comment:
Similar comment here: instead of `assertTrue`, `assertThat` with the `containsString` matcher would be better.